src/utils/dexFormater.js

```diff
@@ -85,47 +102,30 @@ const calculateAmountMin = (
   subResult = true,
 ) => {
   const defaultMinPercent = 0.5;
-  const minAmountValue = (Number(minAmountPercent || defaultMinPercent) / 100);
-  const multiplier = new BN((minAmountValue * 1e18).toString());
-  const { div, mod } = amount.mul(multiplier).divmod(new BN(1e18.toString()));
-  const stringValue = mod.toString();
-
-  const integerPart = stringValue.slice(0, -1e18.toString().length) || '0';
-  const decimalPart = stringValue.slice(-1e18.toString().length);
-
-  const dividedValueString = `${integerPart}.${decimalPart}`;
-  const substract = new BN(Number(div) + Number(dividedValueString));
-  const result = subResult
-    ? amount.sub(substract.isZero() ? new BN(2) : substract)
-    : amount.add(substract.isZero() ? new BN(2) : substract);
-  return result;
-};
-
-export const calculateAmountDesiredFormatted = (desiredAmount, decimals) => {
-  const valueString = desiredAmount.toString();
-  const decimalsFromValue = valueString.split('.')[1]?.length || '0';
-  return new BN(valueString.replace('.', ''))
-    .mul(decimals)
-    .div((new BN(10)
-      .pow(new BN(decimalsFromValue))));
+  const minAmountValue = (Number(minAmountPercent || defaultMinPercent) / 100).toString();
+  const lengthDecimalPercent = minAmountValue.split('.')[1].length;
+  const minAmountPercentBN = parseAssets(minAmountValue, lengthDecimalPercent);
+  const decimalsBN = new BN(10).pow(new BN(lengthDecimalPercent));
+  const percentBN = amount.mul(minAmountPercentBN).divn(decimalsBN);
```

Comment:
Way too overengineered. Just assume a big enough denominator instead of trying to guess it based on string formatting of the fraction. So something like this:
```
const denominator = 10000;
const slippagePercentBN = new BN(minAmountPercent * denominator / 100);
const maxSlippage = amount.mul(slippagePercentBN).div(denominator);
```